internal/expr/bindtypes.go

```diff
@@ -176,50 +191,50 @@ func bindOutputTypes(ti typeNameToInfo, e *outputExpr) (outCols []columnAccessor
 		for _, t := range e.targetTypes {
 			info, ok := ti[t.typeName]
 			if !ok {
-				return nil, nil, typeMissingError(t.typeName, getKeys(ti))
+				return nil, typeMissingError(t.typeName, getKeys(ti))
 			}
 			if t.memberName == "*" {
 				// Generate asterisk columns.
 				allMembers, err := info.GetAllMembers()
 				if err != nil {
-					return nil, nil, err
+					return nil, err
 				}
-				typeMembers = append(typeMembers, allMembers...)
 				for _, tm := range allMembers {
-					outCols = append(outCols, columnAccessor{pref, tm.MemberName()})
+					oc := outputColumn{sql: colString(pref, tm.MemberName()), tm: tm}
+					toe.outputColumns = append(toe.outputColumns, oc)
 				}
 			} else {
 				// Generate explicit columns.
 				tm, err := info.TypeMember(t.memberName)
 				if err != nil {
-					return nil, nil, err
+					return nil, err
 				}
-				typeMembers = append(typeMembers, tm)
-				outCols = append(outCols, columnAccessor{pref, t.memberName})
+				oc := outputColumn{sql: colString(pref, t.memberName), tm: tm}
+				toe.outputColumns = append(toe.outputColumns, oc)
 			}
 		}
-		return outCols, typeMembers, nil
+		return toe, nil
 	} else if numColumns > 1 && starColumns > 0 {
-		return nil, nil, fmt.Errorf("invalid asterisk in columns")
+		return nil, fmt.Errorf("invalid asterisk in columns")
 	}
 
 	// Case 2: Explicit columns, single asterisk type e.g. "(col1, t.col2) AS &P.*".
 	if starTypes == 1 && numTypes == 1 {
 		info, ok := ti[e.targetTypes[0].typeName]
 		if !ok {
-			return nil, nil, typeMissingError(e.targetTypes[0].typeName, getKeys(ti))
+			return nil, typeMissingError(e.targetTypes[0].typeName, getKeys(ti))
 		}
 		for _, c := range e.sourceColumns {
 			tm, err := info.TypeMember(c.columnName)
 			if err != nil {
-				return nil, nil, err
+				return nil, err
 			}
-			typeMembers = append(typeMembers, tm)
-			outCols = append(outCols, c)
+			oc := outputColumn{sql: c.String(), tm: tm}
```

Comment:
[Nit]: Now that I see that the usage of `c.String()` is to output SQL maybe it will be good to rename it to `SQL()`.

Reply:
It is in this case, but this is a parser type, it represents a column that was in the SQLair query so it can contain an asterisk which we would not want the SQL method to return.

Reply:
This call has disappeared now with the `newOutputColumn` method.